compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp

```diff
@@ -29,6 +30,47 @@ static llvm::cl::opt<std::string> clPrintDotGraphToFile(
         "nodes represent roots and the green nodes represent hoisted values."),
     llvm::cl::value_desc("filename"));
 
+template <typename AccessorTy>
+static inline bool isAccessorParameterized(SymbolTable moduleSymbols,
+                                           AccessorTy op) {
+  auto global = moduleSymbols.lookup(op.getGlobalName());
+  if (!global) {
+    return true;
+  }
+  auto utilGlobal = cast<Util::GlobalOpInterface>(global);
+  if (!utilGlobal.getGlobalInitialValue()) {
+    return false;
+  }
+  return !isa<Util::SerializableAttrInterface>(
+      utilGlobal.getGlobalInitialValue());
+}
+
+// Today the only way to interact with a global is with loads, stores, and
+// addresses, and globals are the only way to reference parameters given where
+// const-eval is run today. This is a workaround until we have proper dialect
+// interfaces for detecting whether something is evaluatable at compile time.
+static bool isParameterized(SymbolTable moduleSymbols, Operation *initializer) {
+  WalkResult res = initializer->walk([&](Operation *op) {
+    if (auto accessor = dyn_cast<GlobalLoadOpInterface>(op)) {
```

Comment:
prefer TypeSwitch when doing chains like this to reduce overhead - 3 dyn casts look up the type id 3 times, while a TypeSwitch just does it once